pkg/compare/parsing.go

```diff
@@ -132,25 +136,62 @@ func parseYaml[T any](fsys fs.FS, filePath string, structType *T, fileNotFoundEr
 	return nil
 }
 
-func parseTemplates(templatePaths, functionTemplates []string, fsys fs.FS) ([]*template.Template, error) {
-	var templates []*template.Template
+type ReferenceTemplate struct {
+	*template.Template
+	allowMerge bool
+	config     map[string]string
+}
```

Comment:
when transferring to yaml you can have a complete defined struct with all of the features supported in the per template config, instead of the config + allow merge 
also will be great for  package documentation

Reply:
marking as Not relevant any more, too make sure all comments are resolved